apps/condo/domains/acquiring/schema/RegisterMultiPaymentForVirtualReceiptService.js

```diff
@@ -0,0 +1,215 @@
+
+                // negative to pay value
+                if (Big(amount).lte(0)) {
+                    throw new GQLError({
+                        ...errors.RECEIPT_HAVE_NEGATIVE_TO_PAY_VALUE,
+                    }, context)
+                }
+
+                // Stage 3 Generating payments
+                const formula = await getAcquiringIntegrationContextFormula(context, acquiringIntegrationContext.id)
+                const feeCalculator = new FeeDistribution(formula)
+                const {
+                    type,
+                    explicitFee = '0',
+                    implicitFee = '0',
+                    fromReceiptAmountFee = '0',
+                } = feeCalculator.calculate(amount)
+                const explicitFees = type === 'service' ? {
+                    explicitServiceCharge: String(explicitFee),
+                    explicitFee: '0',
+                } : {
+                    explicitServiceCharge: '0',
+                    explicitFee: String(explicitFee),
+                }
+                const paymentCommissionFields = {
+                    ...explicitFees,
+                    implicitFee: String(implicitFee),
+                    serviceFee: String(fromReceiptAmountFee),
+                }
+                const paymentModel = await Payment.create(context, {
+                    dv: 1,
+                    sender,
+                    amount,
+                    currencyCode,
+                    accountNumber,
+                    period,
+                    context: { connect: { id: acquiringContext.id } },
+                    organization: { connect: { id: acquiringContext.organization } },
+                    recipientBic: bic,
+                    recipientBankAccount: bankAccount,
+                    ...paymentCommissionFields,
+                })
+                const payment = { ...paymentModel, serviceFee: paymentCommissionFields.serviceFee }
+
+                const totalAmount = {
+                    amountWithoutExplicitFee: Big(payment.amount),
+                    explicitFee: Big(payment.explicitFee),
+                    explicitServiceCharge: Big(payment.explicitServiceCharge),
+                    serviceFee: Big(payment.serviceFee),
+                    implicitFee: Big(payment.implicitFee),
+                }
+                const multiPayment = await MultiPayment.create(context, {
+                    dv: 1,
+                    sender,
+                    ...Object.fromEntries(Object.entries(totalAmount).map(([key, value]) => ([key, value.toFixed(2)]))),
+                    currencyCode,
+                    user: { connect: { id: context.authedItem.id } },
+                    integration: { connect: { id: acquiringIntegration.id } },
+                    payments: { connect: [{ id: payment.id }] },
+                    // TODO(DOMA-1574): add correct category
+                    serviceCategory: DEFAULT_MULTIPAYMENT_SERVICE_CATEGORY,
```

Comment:
Why not add correct category here?

Reply:
This one comes from all original implementation of RegisterMultiPaymentService. This is still not implemented in all RegisterPayment** implementations

Reply:
Okay, but its not really good to multiply todos in our service